### PR TITLE
Add Submariner UI for Azure cloud

### PR DESCRIPTION
**Pull request by aswinsuryan**
* Added text boxes to accept azure credentials
* The submariner config will created with the values provide

Fixes: [backlog#25786](https://github.com/stolostron/backlog/issues/25786)

Signed-off-by: Aswin Suryanarayanan <aswinsuryan@gmail.com>

### CODE REVIEW

File: frontend/public/locales/en/translation.json
Comment (sridhargaddam) [inline frontend/public/locales/en/translation.json:2094]: Should this be "Standard_F4s_v2"?

File: frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx
Comment (sridhargaddam) [inline frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx:675]:
```suggestion
                                        'Azure Resource Groups are logical collections of virtual machines, storage accounts, virtual networks, web apps, databases, and/or database servers. Typically, users group related resources for an application, divided into groups for production and non-production.'
```

File: frontend/public/locales/en/translation.json
Comment (aswinsuryan) [reply frontend/public/locales/en/translation.json]: ya that is right, this value was something I used for testing , Changed to  Standard_F4s_v2

**Comment by skitt** [comment]
CI is failing with

```
 [frontend] [tsc] src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx(353,58): error TS2339: Property 'azure' does not exist on type '{ aws?: { instanceType: string; } | undefined; gateways?: number | undefined; }'.
```

**Comment by nyechiel** [comment]
@KevinFCormier can someone from the Console team provide review?

**Review by KevinFCormier — CHANGES_REQUESTED** [review]
@aswinsuryan Looks pretty good. Thanks for including both Submariner developers and someone from `console` squad (me). I have reviewed in terms of best practices for the `console` repo, and I will leave verification of the Submariner business logic to others.

Please note you will need to address the typescript error that is showing up in the `lint-check-unittests-sonarcloud` job.

Once that is fixed, you will probably run into a Quality Gate problem. Code coverage needs to be 70% on all new code, so you may need to add some unit testing.

File: package-lock.json
Comment (aswinsuryan) [reply package-lock.json]: Done.

File: frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx
Comment (aswinsuryan) [reply frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx:690]: I see it generated. now

**Comment by aswinsuryan** [comment]
> @aswinsuryan Looks pretty good. Thanks for including both Submariner developers and someone from `console` squad (me). I have reviewed in terms of best practices for the `console` repo, and I will leave verification of the Submariner business logic to others.
> 
> Please note you will need to address the typescript error that is showing up in the `lint-check-unittests-sonarcloud` job.
> 
> Once that is fixed, you will probably run into a Quality Gate problem. Code coverage needs to be 70% on all new code, so you may need to add some unit testing.

Thanks for the review,  fixed the typescript error, but now it is failing due to another reason  I think is the unit test issue you mentioned. I will work on it.

File: frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx
Comment (nyechiel) [reply frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx:690]: @oafischer (docs) might be able to help.

File: frontend/src/routes/Infrastructure/Clusters/ClusterSets/components/EditSubmarinerConfigModal.tsx
Comment (vthapar) [inline frontend/src/routes/Infrastructure/Clusters/ClusterSets/components/EditSubmarinerConfigModal.tsx:126]: Shouldn't this be `.aws` ?

File: frontend/public/locales/en/translation.json
Comment (vthapar) [inline frontend/public/locales/en/translation.json:156]: Why is this repeated?

**Review by vthapar — CHANGES_REQUESTED** [review]

File: frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx
Comment (oafischer) [reply frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx:690]: I can take a look, just send me a link once the strings are added.

File: frontend/public/locales/en/translation.json
Comment (aswinsuryan) [reply frontend/public/locales/en/translation.json:156]: this is an autogenerated code, it is a key-value pair.

File: frontend/src/routes/Infrastructure/Clusters/ClusterSets/components/EditSubmarinerConfigModal.tsx
Comment (aswinsuryan) [reply frontend/src/routes/Infrastructure/Clusters/ClusterSets/components/EditSubmarinerConfigModal.tsx]: oh ya , good catch , fixed it.

**Comment by KevinFCormier** [comment]
@aswinsuryan Looks great! I am ready to approve once you have an approved review from a teammate and the strings have been reviewed by the doc team. CC @oafischer

**Comment by nyechiel** [comment]
> @aswinsuryan Looks great! I am ready to approve once you have an approved review from a teammate and the strings have been reviewed by the doc team. CC @oafischer

@sridhargaddam @vthapar can you please review/approve when you have some time? thanks!

File: frontend/src/resources/submariner-config.ts
Comment (sridhargaddam) [inline frontend/src/resources/submariner-config.ts:66]: nit: can we use azure? (i.e., azureInstanceType)

**Review by vthapar — DISMISSED** [review]

File: frontend/public/locales/en/translation.json
Comment (oafischer) [inline frontend/public/locales/en/translation.json:161]: _Typically, users group related resources for an application_

What is this trying to say?

File: frontend/public/locales/en/translation.json
Comment (oafischer) [inline frontend/public/locales/en/translation.json:2116]: Change to:

The number of worker nodes used to deploy the Submariner gateway component on the managed cluster. If the value is greater than 1, the Submariner gateway HA is enabled automatically (default 1).

Also, what is HA?

File: frontend/public/locales/en/translation.json
Comment (oafischer) [inline frontend/public/locales/en/translation.json:2118]: _Enter gateway count_ has a prompt but _Instance type_ doesn't. Do they both need a prompt?

File: frontend/public/locales/en/translation.json
Comment (oafischer) [inline frontend/public/locales/en/translation.json:2119]: Not sure how it is in other UI instances but typically we would have the parenthesis before the period. I changed it in an example above but just do whatever keeps it consistent across the UI.

File: frontend/public/locales/en/translation.json
Comment (oafischer) [inline frontend/public/locales/en/translation.json:2123]: Change to:

Set the value to true to enable NAT Traversal and false to disable NAT Traversal (default true).

File: frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx
Comment (oafischer) [inline frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx:710]: -> ...with the following command:

File: frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx
Comment (oafischer) [inline frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx:749]: -> ...with the following command:

File: frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx
Comment (oafischer) [inline frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx:768]: -> ...of the following command:

**Review by oafischer — CHANGES_REQUESTED** [review]
Left a few comments on the strings.

File: frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx
Comment (aswinsuryan) [reply frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx]: done.

File: frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx
Comment (aswinsuryan) [reply frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx]: Done.

File: frontend/public/locales/en/translation.json
Comment (aswinsuryan) [reply frontend/public/locales/en/translation.json:161]: This is not related to Submariner, I am not sure about the context either.

File: frontend/public/locales/en/translation.json
Comment (aswinsuryan) [reply frontend/public/locales/en/translation.json]: Done, HA is high availablity

File: frontend/public/locales/en/translation.json
Comment (aswinsuryan) [reply frontend/public/locales/en/translation.json:2118]: Sorry , I am not clear about this, I don't see a prompt here

File: frontend/public/locales/en/translation.json
Comment (aswinsuryan) [reply frontend/public/locales/en/translation.json]: I made the period at the end for all Submariner strings

File: frontend/public/locales/en/translation.json
Comment (aswinsuryan) [reply frontend/public/locales/en/translation.json]: Done

File: frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx
Comment (aswinsuryan) [reply frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx]: Done,

**Comment by aswinsuryan** [comment]
@oafischer Thanks for your review. I addressed it. Have a query about on of them.

File: frontend/public/locales/en/translation.json
Comment (oafischer) [reply frontend/public/locales/en/translation.json:2118]: The prompt here is "Enter"

File: frontend/public/locales/en/translation.json
Comment (oafischer) [reply frontend/public/locales/en/translation.json:161]: OK, either way the sentences don't make sense so give me a minute and I can provide a suggestion to improve them.

File: frontend/public/locales/en/translation.json
Comment (aswinsuryan) [reply frontend/public/locales/en/translation.json]: oh ok, I added in the prompt in instance type as well,

File: frontend/public/locales/en/translation.json
Comment (aswinsuryan) [reply frontend/public/locales/en/translation.json:161]: When I checked the usages of the string it is used only for baseDomainGroup in submarinerInstall page and the string you mentioned above makes sense. I changed it.

**Comment by aswinsuryan** [comment]
> Left one more suggestion for the Azure Resource Groups string. Not sure who would be the right person to review. Thanks for making the other changes @aswinsuryan!

@oafischer I checked usages for the string and is used for Azure baseDomainGroup and the new string you provided is appropriate. I changed it in all the places where I found it is used

**Comment by oafischer** [comment]
Thanks @aswinsuryan! Approving from a doc/writing perspective.

**Review by oafischer — APPROVED** [review]

**Pull request merged**
